avatars/models.py

```diff
@@ -40,9 +42,15 @@ def set_hash(self):
     def fetch_exchange_image(self):
         auth = HttpNtlmAuth(settings.EXCHANGE_USERNAME, settings.EXCHANGE_PASSWORD)
         url = '{base}/Exchange.asmx/s/GetUserPhoto'.format(base=settings.EXCHANGE_URL)
-        ret = requests.get('{url}?email={email}&size=HR360x360'.format(url=url, email=self.email),
-                           auth=auth)
+        try:
+            ret = requests.get('{url}?email={email}&size=HR360x360'.format(url=url, email=self.email),
```

Comment:
You should include a `timeout=5` argument here, otherwise `requests` can hang forever. Five seconds should be plenty.